torch_geometric/datasets/infection_dataset.py

```diff
@@ -0,0 +1,163 @@
+            (default: :obj:`1`)
+        graph_generator_kwargs (Dict[str, Any], optional): Arguments passed to
+            the respective graph generator module in case it gets automatically
+            resolved. (default: :obj:`None`)
+        seeds (List[int], optional): List of seeds to generate the infection
+            graphs. (default: :obj:`None`)
+        max_path_length (int, List[int]): The maximum shortest path length that
+            a node will not be infected. (default: :obj:`5`)
+        num_infected (List[int], optional): Number of randomly selected
+            infected nodes in the graph. (default: :obj:`50`)
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+            (default: :obj:`None`)
+    """
+    def __init__(
+        self,
+        graph_generator: Union[GraphGenerator, str],
+        num_graphs: int = 1,
+        graph_generator_kwargs: Optional[Dict[str, Any]] = None,
+        seeds: Optional[List[int]] = None,
+        max_path_length: Union[int, List[int]] = 5,
+        num_infected: Union[int, List[int]] = 50,
+        transform: Optional[Callable] = None,
+    ):
+        super().__init__(root=None, transform=transform)
+
+        if seeds is not None:
+            assert len(seeds) == num_graphs
+
+        if isinstance(max_path_length, int):
+            max_path_length = [max_path_length] * num_graphs
+        assert len(max_path_length) == num_graphs
+        assert all(max_path_length) >= 0
+
+        if isinstance(num_infected, int):
+            num_infected = [num_infected] * num_graphs
+        assert len(num_infected) == num_graphs
+        assert all(num_infected) >= 0
+
+        self.graph_generator = GraphGenerator.resolve(
+            graph_generator,
+            **(graph_generator_kwargs or {}),
+        )
+        self.seeds = seeds
+        self.max_path_length = max_path_length
+        self.num_infected = num_infected
+
+        data_list = [
+            self.get_graph(
+                path_length=self.max_path_length[i],
+                num_infected=self.num_infected[i],
+                seed=self.seeds[i] if self.seeds is not None else None)
+            for i in range(num_graphs)
+        ]
+        self.data, self.slices = self.collate(data_list)
+
+    def get_graph(self, path_length: int, num_infected: int,
+                  seed: Optional[int] = None) -> Explanation:
+        import networkx as nx
+        if seed is not None:
+            seed_everything(seed)
+        data = self.graph_generator()
+        num_nodes, edge_index = data.num_nodes, data.edge_index
+        rand_perm = torch.randperm(num_nodes)
+
+        x = torch.zeros((num_nodes, 2), dtype=torch.float32)
+        x[rand_perm[:num_infected], 1] = 1  # Infected
+        x[rand_perm[num_infected:], 0] = 1  # Healthy
```

Comment:
In the original implementation, they have two features, as you implemented. However, what is the advantage of having two features? Should we only use one binary feature?

Reply:
I think it makes sense here to use two features, as otherwise, we need to learn about non-infected nodes via the bias term.